sdk/dotnet/Pulumi/Deployment/GrpcDispatcher.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System.Threading.Tasks;
+using Grpc.Core;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    internal class GrpcDispatcher : IRpcDispatcher
```

Comment:
Why did we have to introduce this combined gRPC interface here but not in https://github.com/pulumi/pulumi/pull/3738?

Reply:
We don't have to... we could introduce two separate interfaces instead. But that's what we eventually came up with Cyrus.

Reply:
On a second look, the reason to merge seems arbitrary. I reverted the change and kept the `Engine` and `Monitor` entities separate.